kanidmd/src/lib/be/idl_arc_sqlite.rs

```diff
@@ -710,6 +716,257 @@ impl<'a> IdlArcSqliteWriteTransaction<'a> {
+        *
+        * So given say:
+        *  [1.0, 1.0, 1.0, 1.0]
+        * We know that the sd_1 will be 1.0. Given:
+        *  [1.0, 1.0, 2.0, 3.0]
+        * We know that it will be ~2.57 (mean 1.75 + sd of 0.82).
+        *
+        * The other factor is number of keys. This is thankfully easy! We have that from
+        * vec.len().
+        *
+        * We can now calculate the index slope. Why is it a slope you ask? Because we
+        * plot the data out on a graph, with "variability" on the y axis, and number of
+        * keys on the x.
+        *
+        * Lets plot our data we just added.
+        *
+        *    |
+        *  4 +
+        *    |
+        *  3 +
+        *    |
+        *  2 +           *  eq_member
+        *    |
+        *  1 +           *  eq_name
+        *    |
+        *    +--+--+--+--+--
+        *       1  2  3  4
+        *
+        * Now, if we were to connect a line from (0,0) to each point we get a line with an angle.
+        *
+        *    |
+        *  4 +
+        *    |
+        *  3 +
+        *    |
+        *  2 +           *  eq_member
+        *    |
+        *  1 +           *  eq_name
+        *    |/---------/
+        *    +--+--+--+--+--
+        *       1  2  3  4
+
+        *    |
+        *  4 +
+        *    |
+        *  3 +
+        *    |
+        *  2 +           *  eq_member
+        *    |        /--/
+        *  1 +    /--/   *  eq_name
+        *    |/--/
+        *    +--+--+--+--+--
+        *       1  2  3  4
+        *
+        * (Look it's ascii art, don't judge.).
+        *
+        * Point is that eq_member is "steeper" and eq_name is "shallower". This is what we call
+        * the "slopeyness" aka the jank of the line, or more precisely, the angle.
+        *
+        * Now we need a way to numerically compare these lines. Since the points could be
+        * anywere on our graph:
+        *
+        *    |
+        *  4 +  *
+        *    |
+        *  3 +         *
+        *    |
+        *  2 +     *
+        *    |
+        *  1 +           *
+        *    |
+        *    +--+--+--+--+--
+        *       1  2  3  4
+        *
+        * While we can see what's obvious or best here, a computer has to know it. So we now
+        * assume that these points construct a triangle, going through (0,0), (x, 0) and (x, y).
+        *
+        *
+        *                Λ│
+        *               ╱ │
+        *              ╱  │
+        *             ╱   │
+        *            ╱    │
+        *           ╱     │
+        *          ╱      │
+        *         ╱       │ sd_1
+        *        ╱        │
+        *       ╱         │
+        *      ───────────┼
+        *         nkeys
+        *
+        * Since this is right angled we can use arctan to work out the degress of the line. This
+        * gives us a value from 1.0 to 90.0 (We clamp to a minimum of 1.0, because we use 0 as "None"
```

Comment:
Not really.

Reply:
So we only care about the ordering, and not by how much? Because if there are two entries where `nkeys` is small but `sd_1` isn't and they're not even close to each other, the angle is still going to look pretty much the same, even though one might be much better than the other.

Reply:
We care abotu the ordering when they are extremely different, but not when the difference is low.

For example - name=foo or uuid=x are both going to yield an idl of length 1, so we don't care which is chosen. We just want to make sure either of those terms are looked at "first" before class=account.